src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+                while ( testIdx < nKmers && test.equals(kmers[testIdx]) ) {
+                    ++testIdx;
+                }
+
+                // if the number of entries with the same kmer is too big
+                if ( testIdx-readIdx > MAX_INTERVALS ) {
+                    // kill 'em
+                    while ( readIdx != testIdx ) {
+                        kmerSet.remove(kmers[readIdx++]);
+                    }
+                }
+                readIdx = testIdx;
+                testIdx += 1;
+            }
+        }
+    }
+
+    /**
+     * Class that acts as a mapper from a stream of reads to a stream of <intervalId,read> pairs.
+     * It knows which breakpoint(s) a read belongs to (if any) by kmerizing the read, and looking up each SVKmer in
+     * a multi-map of SVKmers onto intervalId.
+     */
+    private static final class QNamesForKmersFinder implements Function<GATKRead, Iterator<QNameAndInterval>> {
+        private final HopscotchHashSet<KmerAndInterval> kmerAndIntervalSet;
+        private final Set<Integer> intervalIdSet = new HashSet<>();
+        private final List<QNameAndInterval> qNameAndIntervalList = new ArrayList<>();
+        private final Iterator<QNameAndInterval> emptyIterator = Collections.emptyIterator();
+
+        QNamesForKmersFinder(final HopscotchHashSet<KmerAndInterval> kmerAndIntervalSet) {
+            this.kmerAndIntervalSet = kmerAndIntervalSet;
+        }
+
+        public Iterator<QNameAndInterval> apply(final GATKRead read) {
+            intervalIdSet.clear();
+            SVKmerizer.stream(read.getBases(), SVConstants.KMER_SIZE)
+                    .map( kmer -> kmer.canonical(SVConstants.KMER_SIZE) )
+                    .forEach( kmer -> {
+                        final Iterator<KmerAndInterval> itr = kmerAndIntervalSet.bucketIterator(kmer.hashCode());
+                        while ( itr.hasNext() ) {
+                            final KmerAndInterval kmerAndInterval = itr.next();
+                            if (kmer.equals(kmerAndInterval)) {
+                                intervalIdSet.add(kmerAndInterval.getIntervalId());
+                            }
+                        }
+                    });
+            if (intervalIdSet.isEmpty()) return emptyIterator;
+            qNameAndIntervalList.clear();
+            final String qName = read.getName();
+            intervalIdSet.stream().forEach(intervalId -> qNameAndIntervalList.add(new QNameAndInterval(qName, intervalId)));
+            return qNameAndIntervalList.iterator();
+        }
+    }
+
+    /**
+     * Find <intervalId,fastqBytes> pairs for interesting template names.
+     */
+    private static final class ReadsForQNamesFinder {
+        private final HopscotchHashSet<QNameAndInterval> qNamesSet;
+        private final int nIntervals;
+        private final int nReadsPerInterval;
+        private final List<Tuple2<Integer, List<byte[]>>> fastQRecords = new ArrayList<>();
```

Comment:
Does this have to be a field? It seems to be cleared on each call to `call()` so why not just create a new one as needed in that method? It made me wonder if this was supposed to have mutable state across calls.